pkg/lib/feature/siwe/store_redis.go

```diff
@@ -0,0 +1,83 @@
+package siwe
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	goredis "github.com/go-redis/redis/v8"
+
+	"github.com/authgear/authgear-server/pkg/lib/config"
+	"github.com/authgear/authgear-server/pkg/lib/infra/redis/appredis"
+	"github.com/authgear/authgear-server/pkg/util/clock"
+)
+
+type StoreRedis struct {
+	Redis *appredis.Handle
+	AppID config.AppID
+	Clock clock.Clock
+}
+
+func (s *StoreRedis) Create(nonce *Nonce) error {
+	ctx := context.Background()
+	data, err := json.Marshal(nonce)
+	if err != nil {
+		return err
+	}
+
+	return s.Redis.WithConn(func(conn *goredis.Conn) error {
+		nonceKey := redisNonceKey(s.AppID, nonce)
+		ttl := nonce.ExpireAt.Sub(s.Clock.NowUTC())
+		_, err := conn.Set(ctx, nonceKey, data, ttl).Result()
```

Comment:
Set is different from https://redis.io/commands/setnx/
Set can always set the value so duplicate error will not occur.